src/vehicle_move.cpp

```diff
@@ -586,9 +586,9 @@ void vehicle::thrust( int thd, int z )
         const int min_vel = max_reverse_velocity();
         if( vel_inc > 0 ) {
             // Don't allow braking by accelerating (could happen with damaged engines)
-            velocity = std::max( velocity, std::min( velocity + vel_inc, max_vel ) );
+            velocity = std::min( velocity + vel_inc, max_vel );
         } else {
-            velocity = std::min( velocity, std::max( velocity + vel_inc, min_vel ) );
+            velocity = std::max( velocity + vel_inc, min_vel );
```

Comment:
This possibly changed behavior and might be the cause for the test failures.

Reply:
Good catch. I'm working my way through all the cases, and discovering some places where the existing clamp() or min(max()) use cases have the max lower than the min with unexpected results. I will be either correcting my change or fixing the scenario that leads to those outcomes.